worlds/banjo_tooie/__init__.py

```diff
@@ -0,0 +1,1094 @@
+        # half is useful, half is filler
+        remainder = total - progression
+        useful = ceil(remainder / 2)
+
+        return (useful, remainder - useful)
+
+    def get_jiggies_in_pool(self) -> List[Item]:
+        itempool = []
+
+        if self.options.jingaling_jiggy:
+            # Below give the king a guarentee Jiggy if option is set
+            self.get_location(locationName.JIGGYIH10).place_locked_item(self.create_item(itemName.JIGGY))
+
+        last_level_requirement = max(self.world_requirements.values())
+        if not self.options.open_hag1 and self.options.victory_condition == VictoryCondition.option_hag1:
+            last_level_requirement = max(last_level_requirement, 70)
+
+        # Buffer of 5 progression so that cryptic hints do not consider every jiggy as required,
+        # and so that the spoiler log does not list the absolute worst jiggies as
+        # part of the playthrough.
+        progression_jiggies = min(last_level_requirement + 5, 90)
+
+        # Buffer that is not considered in logic to make the generation faster while making the seed easier.
+        useful_jiggies = ceil((90 - progression_jiggies - 5)/2)\
+            if self.options.replace_extra_jiggies\
+            else 90 - progression_jiggies
+
+        # Some progression jiggies can be placed as locked items, so we don't add them to the pool.
+        if self.options.jingaling_jiggy:
+            progression_jiggies -= 1
+        if not self.options.randomize_jinjos:
+            progression_jiggies -= 9
+
+        # in case preplaced items are over the progression count
+        if progression_jiggies < 0:
+            useful_jiggies += progression_jiggies
+            progression_jiggies = 0
+
+        itempool += [
+            self.create_item(itemName.JIGGY) for i in range(progression_jiggies)
+        ]
+        itempool += [
+            self.create_item(itemName.JIGGY_AS_USEFUL) for i in range(useful_jiggies)
+        ]
+        return itempool
+
+    def get_notes_in_pool(self) -> List[Item]:
+        if not self.options.randomize_notes:
+            return []
+
+        itempool = []
+
+        progression_notes = ceil(max(self.jamjars_siloname_costs.values()) / 5)
+
+        useful_notes, filler_notes = \
+            self.calculate_useful_filler(int(900 / 5), progression_notes)
+
+        taken_by_clefs = 4 * (self.options.extra_trebleclefs_count + all_item_table[itemName.TREBLE].qty)\
+            + 2 * self.options.bass_clef_amount
+
+        progression_notes -= taken_by_clefs
+
+        # in case preplaced items are over the progression count
+        if progression_notes < 0:
+            useful_notes += progression_notes
+            progression_notes = 0
+        if useful_notes < 0:
+            filler_notes += useful_notes
+            useful_notes = 0
+        if filler_notes < 0:
+            warnings.warn("Number of notes that need to be inserted is somehow negative.")
```

Comment:
This should probably just log a warning/error instead of raising one with `warnings`.